packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectHelper.kt

```diff
@@ -328,8 +328,14 @@ internal object RealmObjectHelper {
         }
     }
 
-    @Suppress("UNUSED_PARAMETER")
-    internal fun setList(obj: RealmObjectInternal, col: String, list: RealmList<Any?>) {
-        TODO()
+    @Suppress("unused") // Called from generated code
+    inline fun <reified T : Any> setList(obj: RealmObjectInternal, col: String, list: RealmList<Any?>) {
+        val existingList = getList<T>(obj, col)
+        if (list !is ManagedRealmList || !RealmInterop.realm_equals(existingList.nativePointer, list.nativePointer)) {
```

Comment:
I don't think this check is correct..imagine this usecase, which I would expect to work:

```
realm.write {
  val obj1 = copyToRealm(Person("Mom"))
  val obj2 = copyToRealm(Person("Data))
  obj1.children.add(Person("Mia"))
  obj2.children = obj1.children
}
```

Looks like we are also missing a test for this.

Reply:
I am pretty sure that it is right, and that it is indirectly covered by the test in https://github.com/realm/realm-kotlin/pull/719/files#diff-a9009c4bd0a6af3222c32304364a65fe380a3f5ecdd1e84db0f5356ada531193R278 ... but lets see when I test it explicitly 🙈

Reply:
Reinspecting the test in https://github.com/realm/realm-kotlin/pull/719/files#diff-a9009c4bd0a6af3222c32304364a65fe380a3f5ecdd1e84db0f5356ada531193R278 and it already seem to update a managed object assigning a managed list from object to another list. Isn't it the linked test exactly what you are looking for?

Reply:
Not 100%, there you assign a managed list to an unmanaged object here https://github.com/realm/realm-kotlin/pull/719/files#diff-a9009c4bd0a6af3222c32304364a65fe380a3f5ecdd1e84db0f5356ada531193R287... I was looking for a test that assigned a managed list to an managed object

Reply:
But the object is managed. The `copyToRealm(Sample())` returns a managed object and the assignment is inside the `apply` block on that.

Reply:
Ah yes 🙈 ...I missed a brace there ... Yes, that means the case I was thinking about is tested. Sorry for the confusion.